classes.py

```diff
@@ -0,0 +1,109 @@
+from dataclasses import dataclass
+from exceptions_my import NotStrException, NotIntException
+
+"""
+Библиотека:
+
+- Класс Book:
+   Используем dataclass для создания книги.
+   Атрибуты: book_id, pages, year, author, price. Book_id по умолчанию None
+   присваивается только при добавлении книги в библиотеку.
+   Выполняем валидацию атрибутов при создании книги. Для валидации создаем
+   собственные исключения
+   Реализуем метод сравнения книг по цене.
+
+
+- Класс Library:
+   Хранит книги и автоматически присваивает каждой книге уникальный id.
+   Имеет методы add_book и get_book_info.
+   Поддерживает метод для поиска книг по автору с перегрузкой: можно искать
+   по одному автору или передавать список авторов.
+
+Переопределить методы str в классах для красивого вывода объектов
+
+Примечание: в рамках задания создать два файла: classes.py и main.py.
+В первом будут описаны все классы, во втором классы будут импортированы и
+использованы.
+"""
+
+
+@dataclass(eq=False, order=False)
+class Book:
+    pages: int
+    year: int
+    author: str
+    name: str
+    price: int
+    book_id: int = None
+
+    def __post_init__(self):
+        if not isinstance(self.pages, int):
+            raise NotIntException(self.pages)
+
+        if not isinstance(self.year, int):
+            raise NotIntException(self.year)
+
+        if not isinstance(self.author, str):
+            raise NotStrException(self.author)
+
+        if not isinstance(self.name, str):
+            raise NotStrException(self.name)
+
+        if not isinstance(self.price, int):
+            raise NotIntException(self.price)
+
+    def __str__(self):
+        return (f'{self.author}, {self.name}, {self.year} year, '
+                f'{self.pages} pages, {self.price} USD')
+
+    def __eq__(self, other):
+        return self.price == other.price
+
+    def __ne__(self, other):
+        return self.price != other.price
+
+    def __lt__(self, other):
+        return self.price < other.price
+
+    def __gt__(self, other):
+        return self.price > other.price
+
+    def __le__(self, other):
+        return self.price <= other.price
+
+    def __ge__(self, other):
+        return self.price >= other.price
+
+
+class Library:
+    def __init__(self):
+        self.library = []
+        self.next_id = 1
+
+    def add_book(self, book):
+        book.book_id = self.next_id
+        self.next_id += 1
+        self.library.append(book)
+
+    def get_book_info(self, authors):
```

Comment:
```suggestion
    def get_book_info(self, book_id):
```

and for authors we can create smt like get_books_info(self, authors)